hw_class/hw_7.py

```diff
@@ -0,0 +1,332 @@
+from typing import List, Dict, Union, Tuple, NoReturn
+import json
+import os
+import shutil
+import re
+import string
+from hw_class.exception import NotJSONFileTransferredError
+import time
+
+
+# Задача-1
+# У вас есть список(list) IP адрессов. Вам необходимо создать
+# класс который будет иметь методы:
+# 1) Получить список IP адресов
+# 2) Получить список IP адресов в развернутом виде
+# (10.11.12.13 -> 13.12.11.10)
+# 3) Получить список IP адресов без первых октетов
+# (10.11.12.13 -> 11.12.13)
+# 4) Получить список последних октетов IP адресов
+# (10.11.12.13 -> 13)
+
+
+class IpChangeMaker():
+    def __init__(self, ip: List[str]):
+        self._ip = ip
+
+    def list_ip(self):
+        return self._ip
+
+    def list_ip_reverse(self):
+        list_ip_reverse = ['.'.join(reversed(ip_unit.split("."))) for ip_unit in self._ip]
+        return list_ip_reverse
+
+    def ip_without_first_octet(self):
+        list_ip_without_first_octet = ['.'.join(ip_unit.split(".")[1:]) for ip_unit in self._ip]
+        return list_ip_without_first_octet
+
+    def last_octet_ip(self):
+        list_last_octet_ip = [ip_unit.split(".")[-1] for ip_unit in self._ip]
+        return list_last_octet_ip
+
+
+# Задача-2
+# У вас несколько JSON файлов. В каждом из этих файлов есть
+# произвольная структура данных. Вам необходимо написать
+# класс который будет описывать работу с этими файлами, а
+# именно:
+# 1) Запись в файл
+# 2) Чтение из файла
+# 3) Объединение данных из файлов в новый файл
+# 4) Получить путь относительный путь к файлу
+# 5) Получить абсолютный путь к файлу
+class JsonFileHandler():
+    def __init__(self, *file_names_json: str):
+        json_files_dict = {number: file_name for number, file_name in enumerate(file_names_json, 1)}
+        try:
+            for key in range(1, (len(json_files_dict) + 1)):
+                if ((json_files_dict[key]).split("."))[-1] != "json":
+                    raise NotJSONFileTransferredError(
+                        "The class accepts only json files, the passed values of the class arguments are not json")
```

Comment:
Naming of the file cannot give you confidence that it is a valid json file. I would better add one more method for json validation to catch JSONDecodeError